### ISSUE TITLE
Chunkload issues when entering End

### ISSUE THREAD

**Issue by ehx-v1** [ExpertCoder14/mc-trident-enchantments#2]
So, after setting up the loop function, I entered the End, and all chunks but one were turned into void. I accidentally fell off the edge and ended up in a weird softlock state of falling, but I assume that's normal when you fall into an unloaded chunk. However, assuming that chunks put you back on top once they do load, the chunk had not yet loaded when I was at -15K (which takes a while to reach). At that point, I didn't expect the chunk to load anymore at all, so I closed the game. That world is now pretty much broken.

I'm not 100% sure Trident Enchantments is responsible for that, however giving the way it works I would assume the following: When in the End, Trident Enchantments tried to access 0,0,0 as usual, however it was not loaded due to not being forceloaded in the End. That caused it to retry perpetually, preventing other functions (such as chunk loading) from working. That means this would probably also happen in the Nether.

That or it wasn't an unloaded chunk and I simply stumbled into the air hole Trident Enchantments makes. Unlikely because I would've recieved void damage, which I didn't. Plus, it's a 1x1 column, and what I fell in clearly was at least chunk-sized. And last but not least, I would've landed on the Shulker Box, not fallen out.

You'll probably want to try the following: Make an additional scoreboard that pauses the function (just like `stop trident_main`, called `trident_dimension_transition` or something). Then, when the player is inside a portal block (goes for both Nether and End Portals), trigger an advancement that revokes itself and sets the new scoreboard flag. Another advancement is set to trigger on any dimension transition. It again revokes itself, then calls `trident:setup` (or a variation without the scoreboard initialization), and finally unsets the new scoreboard flag. Not sure if that works, but it's the best I came up with if it's actually a chunk hang.

In the unlikely case it's the Shulker Box hole, I dunno, maybe instead of air use Barriers or something? Since the Shulker Box, from my understanding, doesn't actually need to be openable.

**Comment by ExpertCoder14** [comment]
Yeah, I never meant for this to be used in other dimensions&mdash;it was only a simple experiment when someone asked for it.

I know how to do what you're asking for, and hopefully I can pull something off soon.

**Comment by ehx-v1** [comment]
Fun fact: I was the one who asked on SE. Not sure if anyone else found this yet.

**Comment by ehx-v1** [comment]
And yes, it's kinda pointless in other dimensions, especially the Nether. You can place water in the End, so it will have some marginal use there once #1 is fixed. Still, it's probably not good if there's a complete chunk hang just because a player forgot to leave their trident at home before going to the Nether.

**Comment by ehx-v1** [comment]
Also (sorry, offtopic for the issue), why did you reject my edit that it's possible to combine the enchantments on an anvil? It does work, I tested it! Put Riptide first, use an anvil somewhere where it doesn't rain on you, and congrats, you have a Riptide Loyalty trident. Tested in Survival.

**Comment by ExpertCoder14** [comment]
@ehx-v1 Oh, I didn't notice that it was you who tried to suggest that. I honestly thought it was impossible, guess it's not!

Please don't take the green or red text personally&mdash;what matters is that the post is eventually accurate. I can't reverse the edit to become approved, however what I can do is implement that edit myself.

**Comment by ehx-v1** [comment]
Yup, would be cool. By the way, it's kind of logical that it works - with the lore Riptide setup, there's no actual Riptide on the trident, so no reason for the anvil to block Loyalty and Channeling.
